### PR TITLE
docker: add verification script

### PR DESCRIPTION
**Pull request by guggero**
Fixes https://github.com/lightningnetwork/lnd/issues/4904.

This PR makes the binaries that are built inside the docker container reproducible as well.
In addition to storing the SHA256 hashes of the built binaries inside the container for easy manual verification, we also add a script that verifies those hashes against a public release that contains signatures from our team members.

The script queries GitHub to get the signatures of a release. Therefore this can only be done in an online environment and if the release was already signed by at least one person who's key is in the `scripts/verify-install.sh` script.
Because of this prerequisite the verification script isn't executed automatically on container start!

This can be tested with an image I pushed to my personal fork (signed release is [here](https://github.com/guggero/lnd/releases/tag/v0.12.0-beta.rc6-testbuild4)) with the following commands:
```shell
$ docker pull guggero/lnd:v0.12.0-beta.rc6-testbuild4
$ docker run --rm --entrypoint="" guggero/lnd:v0.12.0-beta.rc6-testbuild4 /verify-install.sh
```

Expected output:
```
Checking if curl is installed... /usr/bin/curl
Checking if jq is installed... /usr/bin/jq
Checking if gpg is installed... /usr/bin/gpg
Checking if lnd is installed... /bin/lnd
Checking if lncli is installed... /bin/lncli
Detected lnd version v0.12.0-beta.rc6-testbuild4 with SHA256 sum 50df8a2541feaeb34834342b8c0d3035060adfcf1a26db88457b565b1ffb9d2e
Detected lncli version v0.12.0-beta.rc6-testbuild4 with SHA256 sum adb1ac08b0b7e4a1934906af49a156a8d25ef23904bc0ed043b42aba4abfddba
Downloading and importing key F4FC70F07310028424EFC20A8E4256593F177720 from https://keybase.io/guggero/pgp_keys.asc
gpg: directory '/root/.gnupg' created
gpg: keybox '/root/.gnupg/pubring.kbx' created
gpg: key 8E4256593F177720: 1 signature not checked due to a missing key
gpg: /root/.gnupg/trustdb.gpg: trustdb created
gpg: key 8E4256593F177720: public key "Oliver Gugger <gugger@gmail.com>" imported
gpg: Total number processed: 1
gpg:               imported: 1
gpg: no ultimately trusted keys found
pub   rsa4096 2018-05-02 [SC] [expires: 2034-04-28]
      F4FC70F07310028424EFC20A8E4256593F177720
uid           [ unknown] Oliver Gugger <gugger@gmail.com>
sub   rsa4096 2018-05-02 [E] [expires: 2034-04-28]
sub   rsa4096 2019-09-25 [E] [expires: 2021-09-24]
sub   rsa2048 2019-09-25 [S] [expires: 2021-09-24]
sub   rsa2048 2019-09-25 [A] [expires: 2021-09-24]

...
<snip>
...


Release v0.12.0-beta.rc6-testbuild4 found with ID 36337458
Downloading manifest-guggero-v0.12.0-beta.rc6-testbuild4.txt.asc

Verifying manifest-guggero-v0.12.0-beta.rc6-testbuild4.txt.asc
Signature for manifest-guggero-v0.12.0-beta.rc6-testbuild4.txt.asc checks out: 
gpg:                using RSA key 6E01EEC9656903B0542B8F1003DB6322267C373B

Verified lnd and lncli hashes against manifest-guggero-v0.12.0-beta.rc6-testbuild4.txt.asc

SUCCESS! Verified lnd and lncli against 1 signature(s).
```

### CODE REVIEW

**Comment by guggero** [comment]
A bonus I forgot to mention is: With the `scripts/verify-install.sh` you'll also be able to verify your local installation of `lnd` and `lncli` against the release signatures!
Assuming of course you have a version installed that is compatible and has signatures from the team, so `v0.12.0-beta.rc6` or later.

File: scripts/release.sh [hunk 1]
```diff
@@ -181,14 +181,21 @@ function build_release() {
     env CGO_ENABLED=0 GOOS=$os GOARCH=$arch GOARM=$arm go build -v -trimpath -ldflags="${ldflags}" -tags="${buildtags}" ${PKG}/cmd/lncli
     popd
 
+    # Add the hashes for the individual binaries as well for easy verification
+    # of a single installed binary.
+    sha256sum "${dir}/"* >> "manifest-$tag.txt"
```
Comment (Roasbeef) [inline scripts/release.sh:186]: 👍

**Review by Roasbeef — COMMENTED** [review]
Very slick! At this point, we're pretty close to having it such that you can download a single version of `lnd` (the entire source tree), then have it be able to automatically use that single install (that's initially verified) to then auto (well still manually user triggered) update and verify any future releases. 

Will do a run locally before merging, but the diff reads well to me.

**Review by Roasbeef — APPROVED** [review]
LGTM 🐝

**Comment by moparisthebest** [comment]
This is the equivalent of asking the malicious binary if it's malicious or not, it absolutely cannot be used to securely verify the docker image isn't malicious, and you really need to stop recommending it as a way to verify anything.

I think the only way to verify the docker image is to sign the layer hashes afterwards, and distribute another tool to verify them.

Why isn't this secure as-is?  Even if you manually verify the script is right, bash could be compromised, even if you verify the binaries inside the docker container are right, the libraries they use could be compromised, or one of ldconfig/ld.so.conf/LD_LIBRARY_PATH could be set to load malicious code into correct binaries.

**Pull request merged**